library/std/src/sys/unix/thread.rs

```diff
@@ -345,6 +345,33 @@ pub fn available_parallelism() -> io::Result<NonZeroUsize> {
                 }
             }
 
+            #[cfg(target_os = "netbsd")]
+            {
+                unsafe {
+                    let set = libc::_cpuset_create();
+                    if !set.is_null() {
+                        let mut count: usize = 0;
+                        if libc::pthread_getaffinity_np(libc::pthread_self(), libc::_cpuset_size(set), set) == 0 {
+                            #[cfg(target_pointer_width = "32")]
+                            const MAXCPUS: u64 = 32;
+                            #[cfg(target_pointer_width = "64")]
+                            const MAXCPUS: u64 = 256;
+                            for i in 0..MAXCPUS {
+                                match libc::_cpuset_isset(i, set) {
+                                    -1 => break,
+                                    0 => continue,
+                                    _ => count = count + 1,
+                                }
+                            }
+                        }
+                        libc::_cpuset_destroy(set);
+                        if count > 0 {
+                            return Ok(NonZeroUsize::new_unchecked(count));
+                        }
+                    }
+                }
+            }
```

Comment:
Since this doesn't need to affect the rest of the code until it hits a final `return`, can the majority of this block be moved out-of-line into a Rust function so the in-line code something like `if let Some(x) = netbsd_available_parallelism() { return Ok(x) };`?

Reply:
I don't know, maybe that's actually harder to read. The platform grouping just feels odd with two mutually-exclusive cfg blocks inside what is already a cfg-if.

Reply:
since these two blocks are not gonna be reusable pieces, would not make much sense to make them as separate calls.

Reply:
Well, there's plenty of other called-once fn in std, they're just usually used as an abstraction boundary (so you don't have to look at both if only one is relevant). I'm not married to that idea, though, I'm mostly just wondering because it is getting... well, has been for some time a bit hard to read std::sys in general.

Reply:
The linux cgroup stuff has been extracted into its own module for example.